client/my-sites/domains/domain-management/edit/registered-domain.jsx

```diff
@@ -69,71 +73,65 @@ const RegisteredDomain = createReactClass( {
 				</Notice>
 			</a>
 		);
-	},
+	}
 
-	getPrivacyProtection() {
-		const {
-				hasPrivacyProtection,
-				privateDomain,
-				privacyAvailable,
-				name,
-				pendingTransfer,
-			} = this.props.domain,
-			{ slug } = this.props.selectedSite,
-			{ translate } = this.props,
-			privacyPath = domainManagementContactsPrivacy( slug, name ),
-			transferPath = domainManagementTransferOut( slug, name );
-		let label;
+	togglePrivacy = () => {
+		const { selectedSite, translate } = this.props;
+		const { privateDomain, name } = this.props.domain;
 
-		if ( ! privacyAvailable ) {
-			return false;
-		}
+		this.setState( { submitting: true } );
 
-		if ( pendingTransfer ) {
-			label = this.getLabel( {
-				status: 'is-warning',
-				icon: 'notice',
-				message: translate( 'Pending Transfer', {
-					context: 'An icon label when domain is pending transfer.',
-				} ),
-			} );
-		} else if ( hasPrivacyProtection ) {
-			if ( privateDomain ) {
-				label = this.getLabel( {
-					status: 'is-success',
-					icon: 'lock',
-					href: privacyPath,
-					message: translate( 'On', {
-						context: 'An icon label when Privacy Protection is enabled.',
-					} ),
-				} );
+		const callback = error => {
+			if ( error ) {
+				this.props.errorNotice( error.message );
 			} else {
-				label = this.getLabel( {
-					status: 'is-warning',
-					icon: 'notice',
-					href: transferPath,
-					message: translate( 'Disabled for Transfer', {
-						context: 'An icon label when Privacy Protection is temporarily disabled for transfer.',
-					} ),
+				this.props.fetchSiteDomains( selectedSite.ID );
```

Comment:
May I suggest dispatching an action here that would update/toggle the redux domain state `privateDomain` here? Right now what happens is we get the notice that the privacy was enabled or disabled and then a few seconds later the toggle button state updates. Here's how it looks on my side - https://cloudup.com/cHKSUgSb9Mc

I guess fetching all domains data is just slow, and since the enable/disable privacy API returns success I think it's safe to assume that it worked?

Reply:
I have a task to make this better. :) Need `fetchSiteDomain`. And we can refresh only that domain which should be way faster, and efficient. That ok with you?

Reply:
What bothers me here is that turning off privacy for WWD domains is asynchronous operation, so if we refresh the status from the domain endpoint it may go back to the previous state (and the user will still be confused). So that's why I think we don't need to make any calls to the API and just change the redux state internally. In that case it is still possible to click on the button and then in a bit (after we refresh the domains state) the privacy will revert back to what it was, but those cases will be a minority if they happen at all... unless there is an issue with removing privacy in which case it doesn't really matter anyway :)

Reply:
Actually, you are right. :) I will remove the refetch, and do what you suggested!